### PR TITLE
Call JaxRsFactoryImplicitBeanCDICustomizer.afterServiceInvoke() to release CreationContexts

### LINKED ISSUE #24155: Memory leak in JaxRsFactoryImplicitBeanCDICustomizer
`CreationalContext`s are not being released when a JAX-RS request returns an error resulting in an `OutOfMemoryError`

This is a follow up to https://github.com/OpenLiberty/open-liberty/issues/18068

### PR DESCRIPTION
**Pull request by WhiteCat22**
This solves a memory leak in `JaxRsFactoryImplicitBeanCDICustomizer` where `CreationContext`s were not released

Fixes: #24155

### CODE REVIEW

**Comment by WhiteCat22** [comment]
Fixes: https://github.com/OpenLiberty/open-liberty/issues/24155

File: dev/com.ibm.ws.jaxrs.2.0.cdi.1.2_fat/fat/src/com/ibm/ws/jaxrs20/cdi12/fat/test/BeanValidation12Test.java [hunk 1]
```diff
@@ -79,4 +80,18 @@ public void testIsViolatedInSingletonWithCDI12_BeanValidation() throws Exception
...
     }
+
+    @Test
+    public void testIsViolatedInPerRequestWithCDI12Leak_BeanValidation() throws Exception {
+
+        for (int i = 0; i < 10000; i++) {
+            runGetMethod("/rest/perrequestleak/book", 400, "I am a Student. null", true);
+        }
+        Thread.sleep(10000);
+        System.gc();
```
Comment (jhanders34) [inline dev/com.ibm.ws.jaxrs.2.0.cdi.1.2_fat/fat/src/com/ibm/ws/jaxrs20/cdi12/fat/test/BeanValidation12Test.java]: The GC needs to happen on the server side, not the test side.
```suggestion
```

File: dev/com.ibm.ws.jaxrs.2.0.cdi.1.2_fat/fat/src/com/ibm/ws/jaxrs20/cdi12/fat/test/BeanValidation12Test.java [hunk 2]
```diff
@@ -79,4 +80,18 @@ public void testIsViolatedInSingletonWithCDI12_BeanValidation() throws Exception
...
+        StringBuilder lines = runGetMethod("/rest/perrequestleak/size", 200, "", false);
+        String result = lines.toString().trim();
+        System.out.println("Adam result=" + result);
+        assertTrue(Integer.valueOf(result) < 10000);
```
Comment (jhanders34) [inline dev/com.ibm.ws.jaxrs.2.0.cdi.1.2_fat/fat/src/com/ibm/ws/jaxrs20/cdi12/fat/test/BeanValidation12Test.java]:
```suggestion
        assertEquals(1, Integer.parseInt(result));
```

File: dev/com.ibm.ws.jaxrs.2.0.cdi.1.2_fat/test-applications/beanvalidation/src/com/ibm/ws/jaxrs20/cdi12/fat/beanvalidation/BookStoreWithValidationPerRequestLeak.java [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+ * Copyright (c) 2023 IBM Corporation and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License 2.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *     IBM Corporation - initial API and implementation
```
Comment (jhanders34) [inline dev/com.ibm.ws.jaxrs.2.0.cdi.1.2_fat/test-applications/beanvalidation/src/com/ibm/ws/jaxrs20/cdi12/fat/beanvalidation/BookStoreWithValidationPerRequestLeak.java]:
```suggestion
```

For new files, you shouldn't add this.

File: dev/com.ibm.ws.jaxrs.2.0.cdi.1.2_fat/test-applications/beanvalidation/src/com/ibm/ws/jaxrs20/cdi12/fat/beanvalidation/BookStoreWithValidationPerRequestLeak.java [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+    @Produces(MediaType.TEXT_PLAIN)
+    public String book(@NotNull @QueryParam("id") String id) {
+        return person.talk() + " " + id;
+    }
+
+    @GET
+    @Path("size")
+    @Produces(MediaType.TEXT_PLAIN)
+    public String book() {
+        int size = map.size();
```
Comment (jhanders34) [inline dev/com.ibm.ws.jaxrs.2.0.cdi.1.2_fat/test-applications/beanvalidation/src/com/ibm/ws/jaxrs20/cdi12/fat/beanvalidation/BookStoreWithValidationPerRequestLeak.java:49]: Here is where you would put the GC logic.
```suggestion
        int size = map.size();
        for (int i = 0; i < 10; ++i) {
            if (size == 1) {
                break;
            }
            System.out.println("calling system gc");
            System.gc();
            try {
                Thread.sleep(1000);
            } catch (Exception e) {
            }
            size = map.size();
        }
```

File: dev/com.ibm.ws.jaxrs.2.0.cdi.1.2_fat/test-applications/beanvalidation/src/com/ibm/ws/jaxrs20/cdi12/fat/beanvalidation/BookStoreWithValidationPerRequestLeak.java [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+import javax.ws.rs.GET;
+import javax.ws.rs.Path;
+import javax.ws.rs.Produces;
+import javax.ws.rs.QueryParam;
+import javax.ws.rs.core.MediaType;
+
+@Path("/perrequestleak/")
+public class BookStoreWithValidationPerRequestLeak {
+
+    static WeakHashMap<BookStoreWithValidationPerRequestLeak, String> map = new WeakHashMap<BookStoreWithValidationPerRequestLeak, String>();
```
Comment (jhanders34) [inline dev/com.ibm.ws.jaxrs.2.0.cdi.1.2_fat/test-applications/beanvalidation/src/com/ibm/ws/jaxrs20/cdi12/fat/beanvalidation/BookStoreWithValidationPerRequestLeak.java]:
```suggestion
    static WeakHashMap<BookStoreWithValidationPerRequestLeak, String> map = new WeakHashMap<>();
```
This is a nit, but might as well do it the simpler way that was added in Java 7.

**Review by jhanders34 — CHANGES_REQUESTED** [review]

File: dev/com.ibm.ws.jaxrs.2.0.common/src/com/ibm/ws/jaxrs20/injection/LibertyClearInjectRuntimeCtxOutInterceptor.java [hunk 6]
```diff
@@ -43,4 +43,12 @@ public void handleMessage(Message message) throws Fault {
             InjectionRuntimeContextHelper.removeRuntimeContext();
         }
     }
+
+    @Override
```
Comment (jhanders34) [inline dev/com.ibm.ws.jaxrs.2.0.common/src/com/ibm/ws/jaxrs20/injection/LibertyClearInjectRuntimeCtxOutInterceptor.java:47]: Copyright update?

File: dev/com.ibm.ws.org.apache.cxf.cxf.rt.frontend.jaxrs.3.2/src/com/ibm/ws/jaxrs20/injection/LibertyClearInjectRuntimeCtxOutInterceptor.java [hunk 7]
```diff
@@ -43,4 +43,12 @@ public void handleMessage(Message message) throws Fault {
             InjectionRuntimeContextHelper.removeRuntimeContext();
         }
     }
+
+    @Override
```
Comment (jhanders34) [inline dev/com.ibm.ws.org.apache.cxf.cxf.rt.frontend.jaxrs.3.2/src/com/ibm/ws/jaxrs20/injection/LibertyClearInjectRuntimeCtxOutInterceptor.java:47]: copyright date update?

**Review by jhanders34 — APPROVED** [review]

**Comment by WhiteCat22** [comment]
#build #libby

**Comment by WhiteCat22** [comment]
network blip in my previous #build

**Pull request merged**